ocs_ci/ocs/platform_nodes.py

```diff
@@ -2418,5 +2417,108 @@ def restart_nodes_by_stop_and_start_teardown(self):
         if len(worker_nodes_not_ready) > 0:
             for not_ready_node in worker_nodes_not_ready:
                 cmd = f"ibmcloud ks worker reboot --cluster {cluster_id} --worker {not_ready_node} -f"
-                out = run_cmd(cmd)
+                out = run_login_cmd(cmd)
                 logger.info(f"Node restart command output: {out}")
+
+    def get_all_workers_status(self, cmd):
+        """
+        Relying on workers status  to confirm that
+        node is added to cluster
+
+        Args:
+            cmd (str): command to get the workers
+
+        Returns:
+            bool: 'True' if all the node names appeared in 'Ready'
+            else 'False'
+
+        """
+        logger.info("Getting all workers status")
+        out = run_cmd(cmd)
+        worker_nodes = json.loads(out)
+        for worker_node in worker_nodes:
+            node_id = worker_node["id"]
+            logger.info(f"{node_id} status is : {worker_node['health']['message']}")
+            if worker_node["health"]["message"] != "Ready":
+                return False
+
+        return True
+
+    def create_nodes(self, node_conf, node_type, num_nodes):
+        """
+        Create new new node by the end of the test on IBM Cloud.
+
+        Args:
+            node_conf (dict): of node configuration
+            node_type (str): type of node to be created RHEL
+            num_nodes (int): Number of node instances to be created
+
+        Returns:
+           list: of IBMCloudNode objects
+
+        Raises:
+           NotAllNodesCreated: In case all nodes are not created
+           TimeoutExpiredError: In case node is not created in time
+
+        """
+        logger.info("creating new node")
+
+        worker_nodes = get_nodes(node_type="worker")
+        provider_id = worker_nodes[0].get()["spec"]["providerID"]
+        cluster_id = provider_id.split("/")[5]
+
+        cmd = f"ibmcloud ks workers --cluster {cluster_id} --output json"
+        out = run_login_cmd(cmd)
+        worker_nodes = json.loads(out)
+
+        no_of_nodes = len(worker_nodes) + int(num_nodes)
+        logger.info(f"number of nodes going to be add are : {no_of_nodes}")
+
+        cmd = (
+            f"ibmcloud ks worker-pool resize --cluster {cluster_id} --worker-pool default"
```

Comment:
This is not really clear to me here.
From help command I see this:
`--size-per-zone value      Specify the desired number of workers per zone in this worker pool.`

Which should be probably calculated based on zones. 
Whe have this function:
https://github.com/red-hat-storage/ocs-ci/blob/1e0d9838cc2cd8f7c8d829531149f208ae3d4a3e/ocs_ci/utility/utils.py#L1760

Which returns the number of zones based on config - or by default 0 if not specified.
We do not specify number of zones in IBM Cloud deployment config at the moment:

https://github.com/red-hat-storage/ocs-ci/blob/1e0d9838cc2cd8f7c8d829531149f208ae3d4a3e/conf/deployment/ibmcloud/ibm_cloud_vpc_cluster.yaml

So I think we have only one zone atm.

In AWS we have config for 3AZ: 
https://github.com/red-hat-storage/ocs-ci/blob/1e0d9838cc2cd8f7c8d829531149f208ae3d4a3e/conf/deployment/aws/ipi_3az_rhcos_3m_3w.yaml#L8-L11

So if I understand the help of resize command it should be worker size per zone - in the case we have 1 zone it will work fine here - but in the case we have 3 AZ then the calculation will need to be per zone and you cannot actually add only one node but it will automatically add 3 worker nodes, 1 in each zone.

Is my assumption correct? If so, the logic here will need to be updated + documentation to cover this as well.

Reply:
nodes will be added based on zones added in worker pool
if worker pool 'default' has zones 'us-south-1' and 'us-south-2'
a)if size-per-zone is  '1',  it will create one node in 'us-south-1' and 'us-south-2'. so total 2 nodes will be created in this worker pool
b) if size-per-zone is '2',  it will create 2 nodes in each 'us-south-1' and 'us-south-2'.    So total 4 nodes will be created in this worker pool

Reply:
Done